test/insert.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+
+
+test('insert', function() {
+
+  eq(typeof S.insert, 'function');
+  eq(S.insert.length, 3);
+  eq(S.insert.toString(), 'insert :: String -> a -> StrMap a -> StrMap a');
+
+  function testBasic(key, val, strMap, strMapResult) {
+    eq(S.insert(key, val, strMap), strMapResult);
+  }
+
+  function unique(sortedArray) {
+    var uniqueArray = [];
+    sortedArray.forEach(function(val, i) {
+      if (i === 0 || val !== sortedArray[i - 1]) {
+        uniqueArray.push(val);
+      }
+    });
+    return uniqueArray;
+  }
+
+  function testKeysOfInsert(key, val, strMap) {
+    eq(
+      unique(S.sort(S.concat(S.keys(strMap), [key]))),
+      S.sort(S.keys(S.insert(key, val, strMap)))
+    );
+  }
+
+  function testDoubleInsert(key, val, strMap) {
+    eq(
+      S.insert(key, val, S.insert(key, val, strMap)),
+      S.insert(key, val, strMap)
+    );
```

Comment:
This might make the double application even clearer:

```javascript
var f = S.insert(key, val);
eq(f(f(strMap)), f(strMap));
```

Reply:
I agree.

Reply:
Would this be clearer?
```javascript
var insert = S.insert(key, val);
eq(insert(insert(strMap)), insert(strMap));
```
I can do the same for `remove`:
```javascript
var remove = S.remove(key);
eq(remove(remove(strMap)), remove(strMap));
```

Reply:
Works for me. :thumbsup:

The name doesn't much matter to me, so long as the pattern is clear. :)